datatypes/src/plots/multi_line_plot.rs

```diff
@@ -0,0 +1,110 @@
+use crate::plots::{Plot, PlotData, PlotMetaData};
+use crate::primitives::{Measurement, TimeInstance};
+use crate::util::Result;
+
+pub struct DataPoint {
+    pub series: String,
+    pub time: TimeInstance,
+    pub value: f64,
+}
+
+impl From<(String, TimeInstance, f64)> for DataPoint {
+    fn from(p: (String, TimeInstance, f64)) -> Self {
+        Self {
+            series: p.0,
+            time: p.1,
+            value: p.2,
+        }
+    }
+}
+
+pub struct MultiLineChart {
+    data: Vec<DataPoint>,
+    measurement: Measurement,
+}
+
+impl MultiLineChart {
+    pub fn new(data: Vec<DataPoint>, measurement: Measurement) -> Self {
+        Self { data, measurement }
+    }
+}
+
+impl Plot for MultiLineChart {
+    fn to_vega_embeddable(&self, _allow_interactions: bool) -> Result<PlotData> {
+        let data = self
+            .data
+            .iter()
+            .map(|d| {
+                serde_json::json!({
+                    "x": d.time.as_rfc3339(),
+                    "y": d.value,
+                    "series": d.series,
+                })
+            })
+            .collect::<Vec<_>>();
+
+        let x_axis_label = "Time";
+        let y_axis_label = self.measurement.to_string();
+
+        let vega_string = serde_json::json!({
+            "$schema": "https://vega.github.io/schema/vega-lite/v4.17.0.json",
+            "data": {
+                "values": data
+            },
+            "description": "Multi Line Chart",
+            "encoding": {
+                "x": {
+                    "field": "x",
+                    "title": x_axis_label,
+                    "type": "temporal"
+                },
+                "y": {
+                    "field": "y",
+                    "title": y_axis_label,
+                    "type": "quantitative"
+                },
+                "color": {
+                    "field": "series",
+                    "scale": {
+                        "scheme": "category20"
+                    }
+                }
+            },
+            "mark": {
+                "type": "line",
+                "line": true,
+                "point": true
+            }
+        })
+        .to_string();
+
+        Ok(PlotData {
+            vega_string,
+            metadata: PlotMetaData::None,
+        })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn serialization() {
+        let chart = MultiLineChart::new(
+            vec![
+                ("S0".to_owned(), TimeInstance::from_millis_unchecked(0), 0.).into(),
+                ("S1".to_owned(), TimeInstance::from_millis_unchecked(0), 2.).into(),
+                ("S0".to_owned(), TimeInstance::from_millis_unchecked(0), 1.).into(),
```

Comment:
Please use a different timestamp for the second data point.

Reply:
745b8bf65d6fcc83aa594ac0759c9d472ed21214